File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int,int> mp;
        for(auto& it:arr){
            mp[it]++;
        }
        unordered_set<int> s;
        for(auto& it:mp){
            s.insert(it.second);
        }

        return mp.size()==s.size();
    }
};

File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer,Integer> mp=new HashMap();
        for(int it:arr){
            mp.put(it, mp.getOrDefault(it, 0) + 1);
        }

        Set<Integer> set=new HashSet();

        for(int it:mp.values()){
            set.add(it);
        }

        return mp.size()==set.size();
    }
}